### PR TITLE
Make CXSMILES writer use the default conformer ID

### LINKED ISSUE #7409: drawing mol with a non zero confID results in bad confID error
**Describe the bug**
Goal is to make a new Chem.Mol of a given conformer in the application of using o3a align to get the best scoring conformer.  I followed the description outlined in the rdkit blog post [https://greglandrum.github.io/rdkit-blog/posts/2023-02-04-working-with-conformers.html](url) and encountered a bug (or misunderstanding) in the visualization of the mol for the different conformers that do not have id=0.

**To Reproduce**
notebook as done in and go to last line to render a new molecule with the conformer of interest:
[https://greglandrum.github.io/rdkit-blog/posts/2023-02-04-working-with-conformers.html](url)

cp = Chem.Mol(esomeprazole,confId=5)
cp

results in 
`ValueError: Bad Conformer Id in Chem.Draw`

**Expected behavior**

The draw mol function does a hard assumption that it must be confId=0, but when plucking conformers out of a set, it would be nice to keep its ID so as to maintain its id origin.  Could the drawing function take the first conformer to draw?  

Alternative drawing tools such as py3dmol will render the conformer correctly since the confId does not seem to be referenced,

### LINKED ISSUE #7414: CXSmiles writer does not use default conformer ID
**Describe the bug**
Here's the reproducible;
```
In [3]: m = Chem.MolFromSmiles('CC |(-0.75,0,;0.75,0,)|')

In [4]: m.GetConformer().SetId(5)

In [5]: Chem.MolToCXSmiles(m)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[5], line 1
----> 1 Chem.MolToCXSmiles(m)

ValueError: Bad Conformer Id
```

**Configuration (please complete the following information):**
 - RDKit version: master and releases
 - OS: all

**Additional context**

This is the root cause of #7409

### PR DESCRIPTION
**Pull request by greglandrum**
Two bugs with a one character change! ;-)

Fixes #7409
Fixes #7414

### CODE REVIEW

**Comment by greglandrum** [comment]
@ptosco this is REALLY short

File: Code/GraphMol/SmilesParse/cxsmiles_test.cpp [hunk 1]
```diff
@@ -1468,3 +1468,13 @@ TEST_CASE("write attachment points") {
...
+
+TEST_CASE("github #7414: CXSmiles writer does not use default conformer ID") {
+  SECTION("basics") {
+    auto m = "CC |(-0.75,0,;0.75,0,)|"_smiles;
+    REQUIRE(m);
+    CHECK(m->getNumConformers() == 1);
+    m->getConformer().setId(5);
+    CHECK(MolToCXSmiles(*m) == "CC |(-0.75,0,;0.75,0,)|");
+  }
+}
```
Comment (bp-kelley) [inline Code/GraphMol/SmilesParse/cxsmiles_test.cpp:1480]: Don't you need two conformers to test this?  From an unpatched rdkit:

```
>>> m = Chem.MolFromSmiles("CC |(-0.75,0,;0.75,0,)|")
>>> m.GetConformer().SetId(5)
>>> Chem.MolToCXSmiles(m)
'CC |(-0.75,0,;0.75,0,)|'
```

Comment (bp-kelley) [reply Code/GraphMol/SmilesParse/cxsmiles_test.cpp:1480]: No, you only need one, but for some reason this molecule isn't triggering the original problem:

```
m = Chem.MolFromSmiles("CC |(-0.75,0,;0.75,0,)|")
m.GetConformer().SetId(5)
m.GetConformer(0)
```

```
ValueError: Bad Conformer Id
```

Comment (bp-kelley) [reply Code/GraphMol/SmilesParse/cxsmiles_test.cpp:1480]: Looks like this bug was introduced after 2023.9.6

**Review by bp-kelley — APPROVED** [review]

File: Code/GraphMol/SmilesParse/cxsmiles_test.cpp (hunk 1 above)
Comment (greglandrum) [reply Code/GraphMol/SmilesParse/cxsmiles_test.cpp:1480]: > Looks like this bug was introduced after 2023.9.6

yeah, it came in with the atropisomer code

**Pull request merged**